ipatests/test_integration/test_commands.py

```diff
@@ -699,3 +699,61 @@ def is_tls_version_enabled(tls_version):
         assert not is_tls_version_enabled('tls1')
         assert not is_tls_version_enabled('tls1_1')
         assert is_tls_version_enabled('tls1_2')
+
+    def test_sss_ssh_authorizedkeys(self):
+        """
+        When there is ssh key is added for IPA user then
+        'sss_ssh_authorizedkeys ipauser' should return ssh key.
+
+        Test for : https://pagure.io/SSSD/sssd/issue/3937
+
+        Steps:
+        1. add ipa user with ssh-key.
+        2. call sss_ssh_authorizedkeys and ssh key should be returned
+        3. check user is able to ssh or not.
+        """
+        user = 'testsshuser'
+        passwd = 'Secret123'
+        user_key = '/tmp/ssh-key'
+        # Create a user with a password
+        tasks.kinit_admin(self.master)
+        tasks.user_add(self.master, user, password=passwd)
+        user_kinit_stdin_text = "{old}\n%{new}\n%{new}\n".format(
+            old=passwd, new=passwd)
+        self.master.run_command(['kinit', user],
+                                stdin_text=user_kinit_stdin_text)
+        tasks.kinit_admin(self.master)
+        tasks.run_command_as_user(
+            self.master, user, ['ssh-keygen', '-f', user_key],
+            stdin_text='\n\n')
+        ssh_pub_key = self.master.get_file_contents('{}.pub'.format(
+            user_key), encoding='utf-8')
+        pem_file = '/tmp/cert.pem'
+        key = '/tmp/private.key'
+        openssl_cmd = [
+            'openssl', 'req', '-x509', '-newkey', 'rsa:2048', '-days', '365',
+            '-nodes', '-keyout', key, '-out', pem_file, '-subj', '/CN=' + user]
+        self.master.run_command(openssl_cmd)
+        cert_b64 = self.get_cert_base64(self.master, pem_file)
+        sssd_p11_child = '/usr/libexec/sssd/p11_child'
+        backup = tasks.FileBackup(self.master, sssd_p11_child)
+        try:
+            content = '#!/bin/bash\nsleep 999999'
+            # added sleep to simulate the timeout for p11_child
+            self.master.put_file_contents(sssd_p11_child, content)
+            self.master.run_command('ipa user-mod {} --ssh "{}"'.format(
+                user, ssh_pub_key))
+            self.master.run_command([
+                'ipa', 'user-add-cert', user, "--certificate=%s" % cert_b64])
+            self.master.run_command(['sss_cache', '-E'])
+            result = self.master.run_command(
+                [paths.SSS_SSH_AUTHORIZEDKEYS, user])
+            assert ssh_pub_key in result.stdout_text
+            # login to the system
+            self.master.run_command(
```

Comment:
Please try avoid interactive communication using stdin. 
Use the following option to ssh:  * `-o StrictHostKeyChecking=no` - do not ask for confirmation if new host is seen and do not fail if host key has changed